hummingbot/connector/gateway_EVM_AMM.py

```diff
@@ -285,43 +306,55 @@ async def get_quote_price(
                 is_buy,
                 amount
             )
+            # Grab the gas price for test net.
+            try:
+                resp: Dict[str, Any] = await GatewayHttpClient.get_instance().get_price(
+                    self.chain, self.network, self.connector_name, base, quote, amount, side
+                )
+                gas_price_token: str = resp["gasPriceToken"]
+                gas_cost: Decimal = Decimal(resp["gasCost"])
+                self.update_transaction_fees(gas_price_token, gas_cost)
+            except asyncio.CancelledError:
+                raise
+            except Exception:
+                pass
+
             if test_price is not None:
                 return test_price
 
         # Pull the price from gateway.
-        base, quote = trading_pair.split("-")
-        side: TradeType = TradeType.BUY if is_buy else TradeType.SELL
         try:
             resp: Dict[str, Any] = await GatewayHttpClient.get_instance().get_price(
                 self.chain, self.network, self.connector_name, base, quote, amount, side
             )
-            required_items = ["price", "gasLimit", "gasPrice", "gasCost"]
+            required_items = ["price", "gasLimit", "gasPrice", "gasCost", "gasPriceToken"]
             if any(item not in resp.keys() for item in required_items):
                 if "info" in resp.keys():
                     self.logger().info(f"Unable to get price. {resp['info']}")
                 else:
                     self.logger().info(f"Missing data from price result. Incomplete return result for ({resp.keys()})")
             else:
-                gas_limit = resp["gasLimit"]
-                gas_price = resp["gasPrice"]
-                gas_cost = resp["gasCost"]
-                price = resp["price"]
-                account_standing = {
-                    "allowances": self._allowances,
-                    "balances": self._account_balances,
-                    "base": base,
-                    "quote": quote,
-                    "amount": amount,
-                    "side": side,
-                    "gas_limit": gas_limit,
-                    "gas_price": gas_price,
-                    "gas_cost": gas_cost,
-                    "price": price,
-                    "swaps": len(resp.get("swaps", []))
-                }
-                exceptions = check_transaction_exceptions(account_standing)
+                gas_limit: int = int(resp["gasLimit"])
+                gas_price_token: str = resp["gasPriceToken"]
+                gas_cost: Decimal = Decimal(resp["gasCost"])
+                price: Decimal = Decimal(resp["price"])
+                self.update_transaction_fees(gas_price_token, gas_cost)
```

Comment:
As per the above suggestion wrt using property.setter, we will have to initialise the new `TokenAmount` here.